explainer.md

```diff
@@ -300,6 +300,18 @@ input.readable.pipeInto(demuxer.writable);
 muxer.readable.pipeInto(output.writable);
 ```
 
+## Detailed design discussion
+
+### Execution environment
+
+Encoder and decoder objects can be instantiated on the main thread and on dedicated workers (TODO: service workers too?). Additionally, the ReadableStream and WritableStream components of each object can be transferred to other contexts (utilizing the [transferrable stream][transferable-streams] infrastructure).
+
+Encode and decode operations can be very computationally expensive. As such, user agents must perform the operations asynchronously with the JavaScript which initiates the operation. The execution environment of the codec implementation is defined by the user agent. Some possibilities include: sequentially or in parallel on an internal thread pool, or on a hardware acceleration unit (e.g., a GPU).
```

Comment:
It's unclear if this is a necessary feature. I could certainly see a mode where this is done synchronously in a specific context. This way, you can have authors decide on the number of threads, parallel, sequential, etc.

Reply:
I'll file a follow-up issue to discuss synchronous execution.